src/components/draw/DrawStyleDirective.js

```diff
@@ -15,16 +15,46 @@ goog.require('ga_window_service');
   module.directive('gaDrawStyle', function($document, $window, $translate,
       gaGlobalOptions, gaStyleFactory, gaUrlUtils, gaWindow) {
 
+    var findCategoryBySource = function(src, categories) {
+      for (var i = 0; i < categories.length; i++) {
+        var c = categories[i];
+        var regex = (c.type === 'img') ?
+          new RegExp('^(.*)' + c.id) :
+          new RegExp(c.regex);
```

Comment:
If regex property is already a RegeEx object you don't have to recreate the regex each time.